lib/internal/Magento/Framework/Lock/LockManagerInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+namespace Magento\Framework\Lock;
+
+/**
+ * Interface of a lock manager
+ *
+ * @api
+ */
+interface LockManagerInterface
+{
+    /**
+     * Sets a lock
+     *
+     * @param string $name lock name
+     * @param int $timeout How long to wait lock acquisition in seconds, negative value means infinite timeout
+     * @return bool
+     * @api
+     */
+    public function acquireLock(string $name, int $timeout = -1): bool;
```

Comment:
Maybe better to have lock($name), unlock($name), tryLock($name, $timeout) methods?

Reply:
I can change to *lock* and *unlock* if there's consensus these are better action verbs. However I would avoid *tryLock* as ambiguous term, for example in Java is has different meaning than just testing the lock status. However *testLock* could be one alternative?